### PR TITLE
Named exports from ol/has

### PR DESCRIPTION
**Pull request by tschaub**
This reworks the `ol/has` module to use named exports.

### CODE REVIEW

**Comment by marcjansen** [comment]
Hey Tim, looking into this now. Please see #7765

File: src/ol/has.js [hunk 1]
```diff
@@ -40,44 +38,14 @@ _ol_has_.MAC = ua.indexOf('macintosh') !== -1;
...
-        if (context.setLineDash !== undefined) {
-          _ol_has_.CANVAS_LINE_DASH = true;
-        }
-        return true;
-      }
-    } catch (e) {
-      return false;
-    }
-  })();
+export const CANVAS_LINE_DASH = false;
```
Comment (marcjansen) [inline src/ol/has.js]: That is quite a change or not?

File: test/spec/ol/interaction/mousewheelzoom.test.js [hunk 2]
```diff
@@ -62,110 +62,106 @@ describe('ol.interaction.MouseWheelZoom', function() {
...
-    describe('spying on ol.interaction.Interaction.zoomByDelta', function() {
-      beforeEach(function() {
-        sinon.spy(Interaction, 'zoomByDelta');
-      });
-      afterEach(function() {
-        Interaction.zoomByDelta.restore();
-      });
-
-      it('works in DOM_DELTA_LINE mode (wheel)', function(done) {
+    if (FIREFOX) {
```
Comment (marcjansen) [inline test/spec/ol/interaction/mousewheelzoom.test.js]: This change is OK for me.

File: test/spec/ol/interaction/mousewheelzoom.test.js [hunk 3]
```diff
@@ -62,110 +62,106 @@ describe('ol.interaction.MouseWheelZoom', function() {
...
         });
         event.coordinate = [0, 0];
         map.handleMapBrowserEvent(event);
       });
+    }
 
-      it('works on Safari (wheel)', function(done) {
-        const origHasSafari = _ol_has_.SAFARI;
-        _ol_has_.SAFARI = true;
+    if (!FIREFOX) {
```
Comment (marcjansen) [inline test/spec/ol/interaction/mousewheelzoom.test.js]: Isn't this too broad? Should this test for Safari explicitly?

**Comment by tschaub** [comment]
Thanks @marcjansen.  I'm debugging the test failure with dashed lines.  It looks like we shouldn't have any conflicts between this and #7765.  I'll review that after fixing up the tests here.

File: src/ol/has.js (hunk 1 above)
Comment (marcjansen) [reply src/ol/has.js]: just read your reply above.

File: test/spec/ol/interaction/mousewheelzoom.test.js (hunk 3 above)
Comment (tschaub) [reply test/spec/ol/interaction/mousewheelzoom.test.js]: This diff view may be making it hard to see, but this condition wraps a test that used to set `FIREFOX = false`.  See https://github.com/openlayers/openlayers/pull/7764/files#diff-7c3aa34c433b5b7112b6f12d7aeea7aaL86.

**Comment by tschaub** [comment]
@marcjansen - I've addressed the previously failing test about line dashes by properly setting `CANVAS_LINE_DASH` in `ol/has`.

**Pull request merged**